src/functions.jl

```diff
@@ -14,8 +14,32 @@ mutable struct ConstraintFunction{F <: Function, G <: Function}
     expression::Expr
 end
 
-function (C::ConstraintFunction)(args...)
-    return C.forward(args)[1]
+function (C::ConstraintFunction)(args...) #To access forward function call f(X) and for backward function call f(A,X)
+                                          #where A is constraint
+
+    if length(args)==1
```

Comment:
You should do this via dispatch instead.

Reply:
okay..!!